Neos.Neos/Classes/Fusion/Cache/ContentCacheFlusher.php

```diff
@@ -163,7 +170,9 @@ public function registerAssetChange(AssetInterface $asset)
                 continue;
             }
 
-            $node = $this->getContextForReference($reference)->getNodeByIdentifier($reference->getNodeIdentifier());
+            $this->securityContext->withoutAuthorizationChecks(function () use ($reference, &$node) {
+                $node = $this->getContextForReference($reference)->getNodeByIdentifier($reference->getNodeIdentifier());
```

Comment:
I guess this change makes sense and I also don't see another way at the moment, but I would prefer to not pull the node out of the closure via reference, that seems hard to understand later on. I would rather pull the code into the closure...

Reply:
This is a bit complicated, than I have to put the whole foreach-loop into the closue. Otherwise I can not `continue` on missing nodes. Or do you have some suggestion?

Reply:
@kitsunet Any (new) opinion on that?

Reply:
@kitsunet Like this? But that makes *everything* without authorization checks…

```php
    public function registerAssetChange(AssetInterface $asset)
    {
        if (!$asset->isInUse()) {
            return;
        }

        $cachingHelper = $this->getCachingHelper();
        $this->securityContext->withoutAuthorizationChecks(function () use ($asset, $cachingHelper) {
            foreach ($this->assetService->getUsageReferences($asset) as $reference) {
                if (!$reference instanceof AssetUsageInNodeProperties) {
                    continue;
                }

                $workspaceHash = $cachingHelper->renderWorkspaceTagForContextNode($reference->getWorkspaceName());
                $node = $this->getContextForReference($reference)->getNodeByIdentifier($reference->getNodeIdentifier());

                if (!$node instanceof NodeInterface) {
                    $this->systemLogger->warning(sprintf('Found a node reference from node with identifier %s in workspace %s to asset %s, but the node could not be fetched.', $reference->getNodeIdentifier(), $reference->getWorkspaceName(), $this->persistenceManager->getIdentifierByObject($asset)), LogEnvironment::fromMethodName(__METHOD__));
                    continue;
                }

                $this->registerNodeChange($node);

                $assetIdentifier = $this->persistenceManager->getIdentifierByObject($asset);
                // @see RuntimeContentCache.addTag
                $tagName = 'AssetDynamicTag_' . $workspaceHash . '_' . $assetIdentifier;
                $this->tagsToFlush[$tagName] = sprintf('which were tagged with "%s" because asset "%s" has changed.', $tagName, $assetIdentifier);
            }
        });
    }
```

Reply:
I added a suggestion with a comment that might help to avoid confusion. I'll merge this later today, unless @kitsunet objects with enough enthusiasm…

Reply:
I guess everfything is fine as long as we fix the bug :D I just wanted to avoid maintenance nightmares